### PR TITLE
[IMPROVEMENT] supporting python 3.6 and flask 1.0.2

### PR DESCRIPTION
**Pull request by satyammittal**
Please prefix your pull request with one of the following: **[FEATURE]** **[FIX]** **[IMPROVEMENT]**.

**In raising this pull request, I confirm the following (please check boxes):**

- [x] I have read and understood the [contributors guide](https://github.com/CCExtractor/sample-platform/blob/master/.github/CONTRIBUTING.md).
- [x] I have checked that another pull request for this purpose does not exist.
- [x] I have considered, and confirmed that this submission will be valuable to others.
- [x] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [x] I give this submission freely, and claim no ownership to its content.

**My familiarity with the project is as follows (check one):**

- [ ] I have never used the project.
- [ ] I have used the project briefly.
- [ ] I have used the project extensively, but have not contributed previously.
- [x] I am an active contributor to the project.

---
a) Fixes taken from port-to-py3
b) Improvements done to support flask 1.0.2 such as session domain and FlaskForm in place of flask_wtf.form
c) "type object not iterable" Error fixed
d) key: expected bytes or bytearray, but got 'str' HMAC issue fixed

after installing python 3.6, dependencies need to be installed again.

### CODE REVIEW

File: .idea/inspectionProfiles/Project_Default.xml [hunk 1]
```diff
@@ -1,7 +1,54 @@
 <component name="InspectionProjectProfileManager">
```
Comment (canihavesomecoffee) [inline .idea/inspectionProfiles/Project_Default.xml]: Could you undo the changes you made into these files? I'm going to look to see what I can do about these files (either remove them, or make them in such a way that they shouldn't be modified).

File: config_parser.py [hunk 2]
```diff
@@ -11,7 +11,7 @@ def parse_config(obj):
     :rtype: dict
     """
     config = {}
-    if isinstance(obj, (str, unicode)):
+    if isinstance(obj, (str)):
```
Comment (canihavesomecoffee) [inline config_parser.py]: Since there's only one item left in the tuple, the tuple can be removed :)

File: config_sample.py [hunk 3]
```diff
@@ -2,7 +2,7 @@
 # install.sh, the config.py should have been generated for you.
 APPLICATION_ROOT = None
 CSRF_ENABLED = True
-DATABASE_URI = 'mysql+pymysql://root:@localhost:3306/test'
+DATABASE_URI = 'mysql+pymysql://root:@localhost:3306/test?charset=utf8'
```
Comment (canihavesomecoffee) [inline config_sample.py]: Is there a reason this needs to be appended? I'd need to know to update the live config :)

File: install/install.sh [hunk 4]
```diff
@@ -27,7 +27,7 @@ echo ""
 echo "* Updating package list"
 apt-get update >> "$install_log" 2>&1
 echo "* Installing nginx, python, pip, kvm, libvirt and virt-manager"
-apt-get -q -y install nginx python python-dev libxslt1-dev libxml2-dev python-pip qemu-kvm libvirt-bin virt-manager mediainfo >> "$install_log" 2>&1
+apt-get -q -y install nginx python python-dev python3-libvirt libxslt1-dev libxml2-dev python-pip qemu-kvm libvirt-bin virt-manager mediainfo python3-libvirt>> "$install_log" 2>&1
```
Comment (canihavesomecoffee) [inline install/install.sh]: Please add a space after the `python3-libvirt` dependency.

File: mod_auth/controllers.py [hunk 5]
```diff
@@ -321,7 +321,7 @@ def complete_signup(email, expires, mac):
     if now <= expires:
         # Validate HMAC
         content_to_hash = "{email}|{expiry}".format(email=email, expiry=expires)
-        real_hash = hmac.new(app.config.get('HMAC_KEY', ''), content_to_hash).hexdigest()
+        real_hash = generate_hmac_hash(app.config.get('HMAC_KEY', ''), content_to_hash).hexdigest()
```
Comment (canihavesomecoffee) [inline mod_auth/controllers.py]: The generate_hmac_hash already returns a hexdigest()... Is this one still needed? The other ones have been removed.

File: mod_auth/controllers.py [hunk 6]
```diff
@@ -360,6 +360,13 @@ def complete_signup(email, expires, mac):
     return redirect(url_for('.signup'))
 
 
+def generate_hmac_hash(key, data):
+    # generate hmac using encoded key and data
```
Comment (canihavesomecoffee) [inline mod_auth/controllers.py]: A better documentation would be appropriate here.

File: mod_auth/forms.py [hunk 7]
```diff
@@ -131,12 +131,12 @@ def validate_password_repeat(form, field):
...
 
 
-class AccountForm(Form):
+class AccountForm(FlaskForm):
     """
     Form for editing current Account
     """
     def __init__(self, formdata=None, obj=None, prefix='', *args, **kwargs):
-        super(AccountForm, self).__init__(formdata, obj, prefix, *args, **kwargs)
+        super(AccountForm, self).__init__(*args, **kwargs)
```
Comment (canihavesomecoffee) [inline mod_auth/forms.py]: Did FlaskForm drop the extra arguments?

Comment (satyammittal) [reply mod_auth/forms.py]: It caused the problem
TypeError: __init__() takes from 1 to 2 positional arguments but 4 were given

File: config_sample.py (hunk 3 above)
Comment (satyammittal) [reply config_sample.py]: while listing users in User Manager, it created some encoding issue while reading db, so i placed utf8 charset there.

**Pull request merged**